### PR TITLE
Support `json_depth`

### LINKED ISSUE #8434: Support `json_depth`

### PR DESCRIPTION
**Pull request by SeaRise**
### What problem does this PR solve?

Issue Number: close #8434

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Functions/FunctionsJson.h [hunk 1]
```diff
@@ -1442,4 +1442,68 @@ class FunctionCastDurationAsJson : public IFunction
...
+            auto & vec_col_res = col_res->getData();
+            vec_col_res.resize(size);
+
+            ColumnString::Offset prev_offset = 0;
+            for (size_t i = 0; i < size; ++i)
+            {
+                size_t data_length = offsets[i] - prev_offset - 1;
+                if (isNullJsonBinary(data_length))
+                {
+                    vec_col_res[i] = 0;
```
Comment (yibin87) [inline dbms/src/Functions/FunctionsJson.h:1489]: Please double check the value for literal null, the value is zero for manually test with tidb.

**Review by yibin87 — APPROVED** [review]
LGTM

**Review by windtalker — APPROVED** [review]
LGTM

File: dbms/src/Functions/FunctionsJson.h [hunk 2]
```diff
@@ -1442,4 +1442,68 @@ class FunctionCastDurationAsJson : public IFunction
...
+                prev_offset = offsets[i];
+            }
+
+            block.getByPosition(result).column = std::move(col_res);
+        }
+        else
+        {
+            throw Exception(
+                fmt::format("Illegal type {} of argument of function {}", from.type->getName(), getName()),
+                ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
```
Comment (Lloyd-Pottiger) [inline dbms/src/Functions/FunctionsJson.h:1505]:
```suggestion
            throw Exception(
                ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT,
                "Illegal type {} of argument of function {}",
                from.type->getName(),
                getName());
```

**Pull request merged**